Sources/Public/Animation/LottieAnimationView.swift

```diff
@@ -225,6 +225,45 @@ final public class LottieAnimationView: LottieAnimationViewBase {
   public var animation: LottieAnimation? {
     didSet {
       makeAnimationLayer(usingEngine: configuration.renderingEngine)
+
+      if let animation = animation {
+        animationLoaded?(self, animation)
+      }
```

Comment:
Can we check whether the animation `!==` to the previous animation before we do this work? Seems like as-is we could potentially call this more than we want

Reply:
Good idea

Reply:
Oh, we don't have access to the previous animation here -- this is a callback for the `animationLoaded` closure itself. The docs say:

> A closure that is called when `self.animation` is loaded. When setting this closure, it is called immediately if `self.animation` is non-nil.

This is a bit non-standard but is required for this pattern to work:

```swift
let animationView: LottieAnimationView

if loadDotLottieFile {
  // Loads the .lottie file asynchronously
  animationView = LottieAnimationView(dotLottieName: "animation")
} else {
  // Loads the .json file synchronously
  animationView = LottieAnimationView(name: "animation")
}

animationView.animationLoaded = { animationView, animation in
  // If using a .lottie file, this is called once the file finishes loading.
  // If using a .json file, this is called immediately (since the animation is loaded synchronously).
  animationView.play()
}
```